lib/jnpr/junos/transport/tty_netconf.py

```diff
@@ -162,10 +163,14 @@ def _receive(self):
             logger.debug('Received: \n%s' % rcvd_data)
             try:
                 etree.XML(rcvd_data)
-            except Exception as ex:
-                if isinstance(ex, etree.XMLSyntaxError):
+            except XMLSyntaxError:
+                if ']]>]]>' in rcvd_data:
                     rcvd_data = rcvd_data[:rcvd_data.index(']]>]]>')]
-                    etree.XML(rcvd_data)
+                    etree.XML(rcvd_data)     # just to recheck
+                else:
+                    parser = etree.XMLParser(recover=True)
+                    rcvd_data = etree.tostring(etree.XML(rcvd_data,
```

Comment:
@stacy because in case of cosole, processing happens with strings and at last the xml string is converted to object and returned back to user.